src/probes/crash_probe.c

```diff
@@ -443,12 +444,18 @@ static bool is_banned_path(char *fullpath)
         }
 
         // Anything outside of /usr/, or in /usr/local/, we consider third-party
-        if (g_regex_match_simple("^[!/]usr[!/]", fullpath, 0, 0) == FALSE) {
-                return true;
-        } else if (g_regex_match_simple("^[!/]usr[!/]local[!/]", fullpath,
-                                        0, 0) == TRUE) {
-                return true;
-        }
+
+	if (strncmp(fullpath, "/usr/", 5) != 0)
```

Comment:
there's only 3 or 4.

a struct is an indirection that will make the code harder to understand; right now the code is linear and you can see how it matches to the intent of the function.
A level of indirection destroys that attempt to be more obvious about what the code does.